### PR TITLE
Split coverlay-minor-mode and global-coverlay-mode

### LINKED ISSUE #10: Please provide a minor mode (or some other means of enabling coverlay per buffer).
Hi!

I like coverlay mode for javascript files, but it only offers a global mode which is superfluous for me in most cases (org files, python files (I use pycoverage for that)). Also it interferes with some of my keybindings (I use C-c C-l for org-insert-link in org files, which is the default binding from org.el). The best solution for my personal taste would be to just enable it for javascript files.

(As a workaround, I'll disable it globally for now and just enable it when I want to see coverage of javascript.)

Thanks!

### PR DESCRIPTION
**Pull request by zonuexe**
resolve #10

```el
'(add-hook 'javascript-mode-hook #'turn-on-coverlay-mode)
```

Cc: @TauPan @orzechowskid

### CODE REVIEW

**Comment by twada** [comment]
@zonuexe Thanks! What a great contribution!

**Comment by twada** [comment]
@TauPan Would you test this pull-req in your environment?

**Comment by TauPan** [comment]
> @TauPan Would you test this pull-req in your environment?
@twada Certainly will, as soon as I find an hour or so. I'll report back. Thanks @zonuexe !

File: coverlay.el [hunk 1]
```diff
@@ -555,38 +554,53 @@
...
+;;;###autoload
+(define-globalized-minor-mode global-coverlay-mode coverlay-minor-mode
+  turn-on-coverlay-mode
+  :group 'coverlay
+  :require 'coverlay)
+
+;;;###autoload
+(make-obsolete 'coverlay-mode #'global-coverlay-mode "2.1.0")
+;;;###autoload
+(make-obsolete-variable 'coverlay-mode-hook 'coverlay-minor-mode-hook "2.1.0")
```
Comment (zonuexe) [inline coverlay.el]: Please release that version (or another appropriate version number) after this PR is merged.

**Comment by TauPan** [comment]
Done testing! Works fine!

FTR I'm using `js2-mode`, so with `use-package` my `.emacs` contains the following
```elisp
(use-package coverlay
  ;; :ensure t
  ;; using from git@github.com:zonuexe/coverlay.el.git/feature/coverlay-minor-mode rightow
  :commands (turn-on-coverlay-mode)
  :init (add-hook 'js2-mode-hook #'turn-on-coverlay-mode))
```

File: coverlay.el (hunk 1 above)
Comment (twada) [reply coverlay.el]: Seems that this PR is a breaking change so it's better to be a version `3.0.0`. Would you update this PR?

File: coverlay.el [hunk 2]
```diff
@@ -555,38 +554,53 @@
...
+;;;###autoload
+(define-globalized-minor-mode global-coverlay-mode coverlay-minor-mode
+  turn-on-coverlay-mode
+  :group 'coverlay
+  :require 'coverlay)
+
+;;;###autoload
+(make-obsolete 'coverlay-mode #'global-coverlay-mode "3.0.0")
+;;;###autoload
+(make-obsolete-variable 'coverlay-mode-hook 'coverlay-minor-mode-hook "3.0.0")
```
Comment (zonuexe) [inline coverlay.el]: Fixed version

File: README.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ CONTRIBUTORS
 * [syohex](https://github.com/syohex)
 * [FossiFoo](https://github.com/fossifoo)
 * [kytwb](https://github.com/kytwb)
-
+* [tadsan](https://github.com/zonuexe)
```
Comment (zonuexe) [inline README.md]: Added my name as contributor.

**Comment by zonuexe** [comment]
@twada https://github.com/twada/coverlay.el/pull/11#discussion_r189173248
I fixed version and squashed some commits.

**Comment by twada** [comment]
@zonuexe Thank you again! Now I'm going to merge this PR.

**Pull request merged**